src/examples/aes/common/aescircuit.h

```diff
@@ -26,30 +26,49 @@
+// on the constant.h definition on the encrypto utils as well.
+// WARNING: Currently a correct running of the algorithms cannot be guaranteed if these values are changed.
+// There might be some work to do.
+#define AES_STATE_KEY_BITS 128
 #define AES_STATE_SIZE_BITS 128
 
-//Size of the expanded key
-#define AES_EXP_KEY_BITS 1408
-#define AES_EXP_KEY_BYTES AES_EXP_KEY_BITS/8
+// The round constant word array, Rcon[i], contains the values given by 
+// x to the power (i-1) being powers of x (x is denoted as {02}) in the field GF(2^8)
+// It is used for the key expansion algorithm
+// Please extend the array if the aes key size changes
+#define RCON_SIZE 11
+const uint8_t Rcon[RCON_SIZE] = {
+  0x8d, 0x01, 0x02, 0x04, 0x08, 0x10, 0x20, 0x40, 0x80, 0x1b, 0x36
+};
+
+#define NB_CONSTANT 4
 
-#define AES_STATE_COLS 4
-#define AES_STATE_ROWS AES_STATE_SIZE/AES_STATE_COLS
 #define INV_GATE_ID 666
 
-const uint8_t AES_TEST_KEY[AES_KEY_BYTES] = { 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 };
+// Derived constants or constants to AES
+#if (AES_STATE_KEY_BITS == 192)
+	#define AES_ROUNDS 12
+	#define AES_STATE_COLS 6
+#elif (AES_STATE_KEY_BITS == 256)
+	#define AES_ROUNDS 14
+	#define AES_STATE_COLS 8
+#else //AES_STATE_KEY__BITS == 128
+	#define AES_ROUNDS 10
+	#define AES_STATE_COLS 4
+#endif
 
-const uint8_t AES_TEST_INPUT[AES_BYTES] = { 0x00, 0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, 0x88, 0x99, 0xaa, 0xbb, 0xcc, 0xdd, 0xee, 0xff };
+#define AES_STATE_SIZE (AES_STATE_SIZE_BITS/8)
 
-const uint8_t AES_TEST_EXPANDED_KEY[AES_EXP_KEY_BITS] = { 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x62, 0x63, 0x63, 0x63,
-		0x62, 0x63, 0x63, 0x63, 0x62, 0x63, 0x63, 0x63, 0x62, 0x63, 0x63, 0x63, 0x9b, 0x98, 0x98, 0xc9, 0xf9, 0xfb, 0xfb, 0xaa, 0x9b, 0x98, 0x98, 0xc9, 0xf9, 0xfb, 0xfb, 0xaa,
-		0x90, 0x97, 0x34, 0x50, 0x69, 0x6c, 0xcf, 0xfa, 0xf2, 0xf4, 0x57, 0x33, 0x0b, 0x0f, 0xac, 0x99, 0xee, 0x06, 0xda, 0x7b, 0x87, 0x6a, 0x15, 0x81, 0x75, 0x9e, 0x42, 0xb2,
-		0x7e, 0x91, 0xee, 0x2b, 0x7f, 0x2e, 0x2b, 0x88, 0xf8, 0x44, 0x3e, 0x09, 0x8d, 0xda, 0x7c, 0xbb, 0xf3, 0x4b, 0x92, 0x90, 0xec, 0x61, 0x4b, 0x85, 0x14, 0x25, 0x75, 0x8c,
-		0x99, 0xff, 0x09, 0x37, 0x6a, 0xb4, 0x9b, 0xa7, 0x21, 0x75, 0x17, 0x87, 0x35, 0x50, 0x62, 0x0b, 0xac, 0xaf, 0x6b, 0x3c, 0xc6, 0x1b, 0xf0, 0x9b, 0x0e, 0xf9, 0x03, 0x33,
-		0x3b, 0xa9, 0x61, 0x38, 0x97, 0x06, 0x0a, 0x04, 0x51, 0x1d, 0xfa, 0x9f, 0xb1, 0xd4, 0xd8, 0xe2, 0x8a, 0x7d, 0xb9, 0xda, 0x1d, 0x7b, 0xb3, 0xde, 0x4c, 0x66, 0x49, 0x41,
-		0xb4, 0xef, 0x5b, 0xcb, 0x3e, 0x92, 0xe2, 0x11, 0x23, 0xe9, 0x51, 0xcf, 0x6f, 0x8f, 0x18, 0x8e };
+// Size of the key
+#define AES_STATE_KEY_BYTES (AES_STATE_KEY_BITS/8)
 
+// Size of the expanded key
+#define AES_EXP_KEY_BYTES (AES_STATE_KEY_BYTES*(AES_ROUNDS+1))
+#define AES_EXP_KEY_BITS (AES_EXP_KEY_BYTES*8)
+
+#define AES_STATE_ROWS (AES_STATE_SIZE/AES_STATE_COLS)
 
```

Comment:
Is there a reason to use lots of `#define`s instead of regular constants?

Reply:
Good question. I didn't think about it yet since the defines were just used before I changed the code. However I don't think it's such a breaking issue so that it needs to be changed again.